TestCertificateController is flaky

This test fails sporadically in CI:

```
=== RUN   TestCertificateController
=== RUN   TestCertificateController/creates_new_configmap_on_pod_add
time="2018-08-08T00:05:46Z" level=info msg="waiting for caches to sync"
time="2018-08-08T00:05:46Z" level=info msg="caches synced"
time="2018-08-08T00:05:46Z" level=info msg="starting certificate controller"
--- FAIL: TestCertificateController (0.15s)
    --- FAIL: TestCertificateController/creates_new_configmap_on_pod_add (0.15s)
    	controller_test.go:54: expected action to be configmap create, got: {ActionImpl:{Namespace: Verb:watch Resource:{Group:apps Version:v1beta2 Resource:replicasets} Subresource:} WatchRestrictions:{Labels: Fields: ResourceVersion:}}
time="2018-08-08T00:05:46Z" level=info msg="shutting down certificate controller"
FAIL
FAIL	github.com/linkerd/linkerd2/controller/ca	0.266s
```

--- a/controller/ca/controller_test.go
+++ b/controller/ca/controller_test.go
@@ -47,16 +47,15 @@ func TestCertificateController(t *testing.T) {
 			t.Fatal("timed out waiting for sync")
 		}
 
-		actions := controller.k8sAPI.Client.(*fake.Clientset).Actions()
-		action := actions[len(actions)-2] // configmap create
-
-		if !action.Matches("create", "configmaps") {
-			t.Fatalf("expected action to be configmap create, got: %+v", action)
+		var found bool
+		for _, a := range controller.k8sAPI.Client.(*fake.Clientset).Actions() {
+			if a.Matches("create", "configmaps") && a.GetNamespace() == injectedNS {
+				found = true
+			}
 		}
 
-		if action.GetNamespace() != injectedNS {
-			t.Fatalf("expected action to happen in [%s] namespace, got [%s] namespace",
-				injectedNS, action.GetNamespace())
+		if !found {
+			t.Fatalf("configmap create event not found in [%s] namespace", injectedNS)
 		}
 	})
 }

